etc/buildspec.yml

```diff
@@ -0,0 +1,20 @@
+version: 0.2
+
+env:
+  variables:
+    POETRY_VIRTUALENVS_CREATE: false
+
+phases:
+  install:
+    runtime-versions:
+      python: 3.10
+    commands:
+      - pip install poetry==1.4.0
+      - poetry install --with dev --no-interaction
+
+  build:
+    commands:
+      - python -m pytest .
+      - python -m black . --check
+      - python -m isort . --check
+      - python -m flake8 .
```

Comment:
Flake8 might not be compatible with black in some cases: https://black.readthedocs.io/en/stable/guides/using_black_with_other_tools.html#flake8, just pointing this out as a potential issue in the future.